### PR TITLE
refactor: Expense Claim Cost Center

### PR DESCRIPTION
**Pull request by jiloysss**
* There is a cost center field added to the table (Expense Claim Detail)
* Changed Script in Expense Claim (python) where the cost center in GL Entry is based on the cost center field on each row (Expense Claim Detail) 
* Filtered Cost Center in Expense Claim Detail where it will only fetch child account
* Hid Cost Center in Expense Claim
* Added Deferred Expense Accounts checkbox in Expense Claim Type

Please read the pull request checklist to make sure your changes are merged: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

### CODE REVIEW

**Comment by jiloysss** [comment]
EXPENSE CLAIM
Added field Cost Center in table

expense_claim
EXPENSE CLAIM TYPE
Added check box Deferred Expense Accounts
- If checked, Default Account in table will be filtered with Asset Accounts. Expense Accounts if otherwise

Expense_claim_type

**Review by sammish — APPROVED** [review]

**Comment by deepeshgarg007** [comment]
@jiloysss Can you tell us a use case for this. Why do you need Asset Accounts in `Expense Claim Type` and Cost Center on each expenses.

**Comment by sammish** [comment]
@deepeshgarg007 **Expense Claim Type:** we can claim **Prepaid expense** via expense claim module. (Debit in **Prepaid asset account** Credit in the ****bank or cash**) this is more flexible to HR Department claim their  **Prepaid expense** without purchase invoice.
**Cost Center on each Expenses:** it will help Employee to claim multiple cost center  expense via single expense claim get approval from management.
for example : Employee  business trip for both projects ( project1 and project2). He got expense 100. Here he can share that amount to **Projcet1** 50 and **Project2** 50 from single expense claim.

**Comment by deepeshgarg007** [comment]
> @deepeshgarg007 **Expense Claim Type:** we can claim **Prepaid expense** via expense claim module. (Debit in **Prepaid asset account** Credit in the ****bank or cash**) this is more flexible to HR Department claim their **Prepaid expense** without purchase invoice.

How will you select Cash or Bank Account in Expense Claim only Payable Accounts are allowed in Explain Claim. There is already a mechanism available for this use case.

1. Create Employee Advance
2. Create Payment Entry against Employee Advance
3. You can then use that Employee Advance to allocate it against a Expense Claim

 
> **Cost Center on each Expenses:** it will help Employee to claim multiple cost center expense via single expense claim get approval from management.
> for example : Employee business trip for both projects ( project1 and project2). He got expense 100. Here he can share that amount to **Projcet1** 50 and **Project2** 50 from single expense claim.

Why the parent cost center at parent level is hidden. And cost center in expenses table should not be mandatory. What if someone does n't want to tag expense wise cost centers.

**Comment by sammish** [comment]
@deepeshgarg007  **How will you select Cash or Bank Account in Expense Claim only Payable Accounts are allowed in Explain Claim. There is already a mechanism available for this use case.** ////  we will not select payable account  like **cash** or **bank** we will select **"Prepaid Account**" under Asset.
image

**Employee advance** is another scenario I am not talking about that. I am talking about **Prepaid Expense** I have well  a knowledge of between **employee advance & Prepaid Expense**


**Why the parent cost center at parent level is hidden. And cost center in expenses table should not be mandatory. What if someone does n't want to tag expense wise cost centers.** ////// current expense cost center also not support **Parent Cost Center**. regarding **Cost center mandatory** if you assign company master default cost center it will not ask expense claim.

File: erpnext/hr/doctype/expense_claim/expense_claim.js [hunk 1]
```diff
@@ -341,4 +342,12 @@ cur_frm.fields_dict['task'].get_query = function(doc) {
...
 		}
 	};
-};
\ No newline at end of file
+};
+
+cur_frm.fields_dict["expenses"].grid.get_field("cost_center").get_query = function(){
+   return {
+	   filters:{
+		   "is_group": "child"
```
Comment (nabinhait) [inline erpnext/hr/doctype/expense_claim/expense_claim.js:350]: "is_group": 0

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/expense_claim/expense_claim.js (hunk 1 above)
Comment (sammish) [reply erpnext/hr/doctype/expense_claim/expense_claim.js:350]: @nabinhait I will Update. Kindly focus version 12 Pull request https://github.com/frappe/erpnext/pull/19326

**Pull request closed without merging**